packages/nx/src/project-graph/nx-deps-cache.ts

```diff
@@ -182,9 +184,11 @@ export function createProjectFileMapCache(
   nxJson: NxJsonConfiguration<'*' | string[]>,
   packageJsonDeps: Record<string, string>,
   fileMap: FileMap,
-  tsConfig: { compilerOptions?: { paths?: { [p: string]: any } } }
+  tsConfig: { compilerOptions?: { paths?: { [p: string]: any } } },
+  externalNodes: Record<string, ProjectGraphExternalNode>
 ) {
   const nxJsonPlugins = getNxJsonPluginsData(nxJson, packageJsonDeps);
+  const externalNodesHash = hashObject(getExternalNodesData(externalNodes));
```

Comment:
Are we hashing it twice? One to compare and once to write it? We should only hash it once.